docs_samples/01_python_app_insights_overview.md

```diff
@@ -0,0 +1,239 @@
+```
+
+- Open the Azure Portal
+- Create a new resource group
+- Run the `./Artifacts/template.json` file
+
+## Configure the environment (Azure)
+
+- In the Azure Portal, browse to the `python-insights-SUFFIX` Azure Database for SQL
+- Under **Security**, select **Networking**
+- Under **Firewall rules**, select to **Add your client IPv4 address(xx.xx.xx.xx)**
+- Check the **Allow Azure services and resources to access this server**
+- Select **Save**
+
+## Configure the environment (Visual Studio Code)
+
+- Open Visual Studio code, select the **Extensions** tab
+- Ensure the following extensions are installed:
+  - ms-python.python
+  - ms-azuretools.vscode-azurefunctions
+
+## Check your Python Version
+
+- In Visual Studio Code, open a Terminal window, run the following command:
+
+```powershell
+python --version
+
+pip install --upgrade pip
+```
+
+- For Python, make sure you see version 3.9.x.  If you do not, do the following:
+  - Download Python and change your path variables to point to the proper version
+  - Restart Visual Studio Code
+- Open the command palette by selecting **View->Command Pallet**
+- Run the `Python: Select interperter` command
+- Select the `Python 3.9.x` version
+
+## Create a Python environment
+
+- Run the following command to create an environment where you can load all the dependencies:
+
+```powershell
+py -3 -m venv .venv
+.venv\scripts\activate
+```
+
+- When prompted, select **yes**
+- Ensure that you select the new environment in the interpertor selection
+
+## Install OpenCensus
+
+[OpenCensus](https://opencensus.io/) is a set of open source libraries hosted on [GitHub](https://github.com/census-instrumentation) that are used to capture trace and metric data from applications.
+
+Metrics can be things like latency, HTTP request and reponse lengths whereas traces are how a request moves through your application layers and other services. Graphs can be built to show how an application or service is performing based on these data points.
+
+To install OpenCensus:
+
+- Open a terminal window, run the following:
+
+```powershell
+pip install --upgrade pip
+
+python -m pip install opencensus
+```
+
+## Install Azure exporter
+
+OpenCensus needs an exporter to know how to send the log and metric data to the target.  Azure provides an exporter for Azure Monitor.
+
+To install the Azure Exporter:
+
+- Open a terminal window, run the following:
+
+```powershell
+python -m pip install opencensus-ext-logging
+python -m pip install opencensus-ext-azure
+python -m pip install opencensus-extension-azure-functions
+python -m pip install opencensus-ext-requests
+python -m pip install pyodbc
+python -m pip install psutil
+
+```
+
+## Configure middleware to track requests
+
+- Browse to the Azure Portal
+- Select your Azure Monitor resource
+- On the **Overview** page, copy the connection string
+- Open the `.env` file and paste it into the `APPLICATIONINSIGHTS_CONNECTION_STRING` value
```

Comment:
These steps that involve a specific file are currently not clear to me. Are they generated  from``./Artifacts/template.json` above?

Reply:
I updated this to be automated now as part of a setup.ps1 file.  People won't really need to do anything except press F5 to run things!